cmplibrary/src/main/java/com/sourcepoint/cmplibrary/SpConsentLibImpl.kt

```diff
@@ -415,12 +418,26 @@ internal class SpConsentLibImpl(
     }
 
     override fun dispose() {
+        webview = null
         executor.dispose()
         viewManager.removeAllViews()
     }
 
     private fun logMess(mess: String) = pLogger.d(this::class.java.simpleName, mess)
 
+    override fun onBackPressed() {
+        webview.let {
+            (webview as ConsentWebView).evaluateJavascript(
+                """
+                window.postMessage({ name: 'sp.BACK' })
+                """.trimIndent(),
+                null
+            )
+        }
+    }
+
+    override fun isWebviewShown(): Boolean = (webview as? ConsentWebView)?.isShown ?: false
```

Comment:
Also, is there a way to implement this logic without checking if the webview is shown?

Reply:
@andresilveirah not sure about that, since we have to prohibit activity's `onBackPressed()` when the `webView` is shown and vice versa, so we need a flag that indicates whether the `webView` is shown or not

Reply:
Calling the onBackPressed from SDK means that we have a reference back to the host, our SDK is a simple library which is just providing 2 APIs, regardless of the host component and we don't want to interfere with the lifecycle of the host object (in our case the Activity or, in a different scenario, a simple view).
This fix won't work if the host object is a fragment or a tv view component